lib/portlayer/exec/container.go

```diff
@@ -377,15 +382,23 @@ func (c *Container) Signal(ctx context.Context, num int64) error {
 	return c.startGuestProgram(ctx, "kill", fmt.Sprintf("%d", num))
 }
 
-func (c *Container) LogReader(ctx context.Context) (io.ReadCloser, error) {
+func (c *Container) stopped() {
```

Comment:
ok onStop it is.  Why would 'onStop' evaluate the state?  eventCallback could probably be refactored a bit, but that's where we are already evaluating state changes and dispatching based on the state change.